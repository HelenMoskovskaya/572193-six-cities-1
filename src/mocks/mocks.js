export const mockOffers = [
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 51.225402,
        longitude: 6.776314,
        zoom: 13
      }
    },
    previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
    images: [
      `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`
    ],
    title: `Waterfront with extraordinary view`,
    isFavorite: false,
    isPremium: true,
    rating: 2.7,
    type: `hotel`,
    bedrooms: 2,
    maxAdults: 8,
    price: 393,
    goods: [
      `Breakfast`,
      `Washer`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
    location: {
      latitude: 51.214402,
      longitude: 6.764314000000001,
      zoom: 16
    },
    id: 1
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 51.225402,
        longitude: 6.776314,
        zoom: 13
      }
    },
    previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
    images: [
      `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`
    ],
    title: `Waterfront with extraordinary view`,
    isFavorite: false,
    isPremium: true,
    rating: 2.7,
    type: `hotel`,
    bedrooms: 2,
    maxAdults: 8,
    price: 393,
    goods: [
      `Breakfast`,
      `Washer`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
    location: {
      latitude: 51.214402,
      longitude: 6.764314000000001,
      zoom: 16
    },
    id: 3
  }
];

export const mockOffer = {
  city: {
    name: `Amsterdam`,
    location: {
      latitude: 51.225402,
      longitude: 6.776314,
      zoom: 13
    }
  },
  previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
  images: [
    `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/8.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/7.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`
  ],
  title: `Waterfront with extraordinary view`,
  isFavorite: false,
  isPremium: true,
  rating: 2.7,
  type: `hotel`,
  bedrooms: 2,
  maxAdults: 8,
  price: 393,
  goods: [
    `Breakfast`,
    `Washer`,
    `Laptop friendly workspace`
  ],
  host: {
    id: 25,
    name: `Angelina`,
    isPro: true,
    avatarUrl: `img/avatar-angelina.jpg`
  },
  description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
  location: {
    latitude: 51.214402,
    longitude: 6.764314000000001,
    zoom: 16
  },
  id: 1
};

export const cityMock = {
  name: `Amsterdam`,
  location: {
    latitude: 51.225402,
    longitude: 6.776314,
    zoom: 13
  }
};

export const cities = [`Amsterdam`, `Paris`, `Colone`];

export const userData = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`,
  isPro: false
};

export const mockFavorites = {
  "Amsterdam": mockOffers,
  "Colone": mockOffers
};

export const reviewsMock = [
  {
    id: 1,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatarUrl: `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  }
];

export const reviewItemMock = {
  id: 1,
  user: {
    id: 4,
    isPro: false,
    name: `Max`,
    avatarUrl: `img/1.png`
  },
  rating: 4,
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`
};
