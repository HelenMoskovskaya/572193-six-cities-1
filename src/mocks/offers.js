export default [
  {
    city: `Paris`,
    centerCityCoords: [48.85881005, 2.32003101],
    offerCoords: [48.83832557, 2.29878187],
    title: `Wood and stone place`,
    type: `Private room`,
    image: `img/room.jpg`,
    price: 80,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Paris`,
    centerCityCoords: [48.85881005, 2.32003101],
    offerCoords: [48.86905515, 2.36973166],
    title: `Wood and stone place`,
    type: `Private room`,
    image: `img/room.jpg`,
    price: 80,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Cologne`,
    centerCityCoords: [50.938361, 6.959974],
    offerCoords: [50.92812784, 6.9745481],
    title: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    image: `img/apartment-03.jpg`,
    price: 180,
    rating: 100,
    isPremium: true,
  },
  {
    city: `Cologne`,
    centerCityCoords: [50.938361, 6.959974],
    offerCoords: [50.94851179, 6.9483161],
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    image: `img/apartment-02.jpg`,
    price: 132,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Brussels`,
    centerCityCoords: [50.8465573, 4.351697],
    offerCoords: [50.82894742, 4.37661409],
    title: `Wood and stone place`,
    type: `Private room`,
    image: `img/room.jpg`,
    price: 80,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Amsterdam`,
    centerCityCoords: [52.38333, 4.9],
    offerCoords: [52.3909553943508, 4.85309666406198],
    title: `Beautiful luxurious apartment at great location`,
    type: `Apartment`,
    image: `img/apartment-01.jpg`,
    price: 120,
    rating: 93,
    isPremium: true,
  },
  {
    city: `Amsterdam`,
    centerCityCoords: [52.38333, 4.9],
    offerCoords: [52.3909553943508, 4.929309666406198],
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    image: `img/apartment-02.jpg`,
    price: 132,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Hamburg`,
    centerCityCoords: [53.550341, 10.000654],
    offerCoords: [53.55453394, 10.01213372],
    title: `Beautiful luxurious apartment at great location`,
    type: `Apartment`,
    image: `img/apartment-01.jpg`,
    price: 120,
    rating: 93,
    isPremium: true,
  },
  {
    city: `Dusseldorf`,
    centerCityCoords: [51.2254018, 6.7763137],
    offerCoords: [51.22621102, 6.78346485],
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    image: `img/apartment-02.jpg`,
    price: 132,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Dusseldorf`,
    centerCityCoords: [51.2254018, 6.7763137],
    offerCoords: [51.23314105, 6.77579641],
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    image: `img/apartment-02.jpg`,
    price: 132,
    rating: 80,
    isPremium: false,
  },
];
