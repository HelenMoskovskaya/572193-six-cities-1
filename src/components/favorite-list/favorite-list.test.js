import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import FavoriteList from './favorite-list.jsx';
import {mockFavorites} from '../../mocks/mocks.js';


const renderer = new ShallowRenderer();

describe(`FavoriteList`, () => {
  it(`renders correctly`, () => {
    renderer.render(
        <FavoriteList
          favorites={mockFavorites}
        />
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
