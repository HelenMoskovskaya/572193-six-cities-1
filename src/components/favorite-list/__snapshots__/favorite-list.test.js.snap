// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteList renders correctly 1`] = `
<ul
  className="favorites__list"
>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="#"
        >
          <span>
            Amsterdam
          </span>
        </a>
      </div>
    </div>
    <div
      className="favorites__places"
    >
      <OfferList
        className="favorites__places"
        favoriteClass={true}
        needLink={true}
        offers={
          Array [
            Object {
              "bedrooms": 2,
              "city": Object {
                "location": Object {
                  "latitude": 51.225402,
                  "longitude": 6.776314,
                  "zoom": 13,
                },
                "name": "Amsterdam",
              },
              "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
              "goods": Array [
                "Breakfast",
                "Washer",
                "Laptop friendly workspace",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 1,
              "images": Array [
                "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
              ],
              "isFavorite": false,
              "isPremium": true,
              "location": Object {
                "latitude": 51.214402,
                "longitude": 6.764314000000001,
                "zoom": 16,
              },
              "maxAdults": 8,
              "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
              "price": 393,
              "rating": 2.7,
              "title": "Waterfront with extraordinary view",
              "type": "hotel",
            },
            Object {
              "bedrooms": 2,
              "city": Object {
                "location": Object {
                  "latitude": 51.225402,
                  "longitude": 6.776314,
                  "zoom": 13,
                },
                "name": "Amsterdam",
              },
              "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
              "goods": Array [
                "Breakfast",
                "Washer",
                "Laptop friendly workspace",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 3,
              "images": Array [
                "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
              ],
              "isFavorite": false,
              "isPremium": true,
              "location": Object {
                "latitude": 51.214402,
                "longitude": 6.764314000000001,
                "zoom": 16,
              },
              "maxAdults": 8,
              "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
              "price": 393,
              "rating": 2.7,
              "title": "Waterfront with extraordinary view",
              "type": "hotel",
            },
          ]
        }
        small={true}
      />
    </div>
  </li>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="#"
        >
          <span>
            Colone
          </span>
        </a>
      </div>
    </div>
    <div
      className="favorites__places"
    >
      <OfferList
        className="favorites__places"
        favoriteClass={true}
        needLink={true}
        offers={
          Array [
            Object {
              "bedrooms": 2,
              "city": Object {
                "location": Object {
                  "latitude": 51.225402,
                  "longitude": 6.776314,
                  "zoom": 13,
                },
                "name": "Amsterdam",
              },
              "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
              "goods": Array [
                "Breakfast",
                "Washer",
                "Laptop friendly workspace",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 1,
              "images": Array [
                "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
              ],
              "isFavorite": false,
              "isPremium": true,
              "location": Object {
                "latitude": 51.214402,
                "longitude": 6.764314000000001,
                "zoom": 16,
              },
              "maxAdults": 8,
              "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
              "price": 393,
              "rating": 2.7,
              "title": "Waterfront with extraordinary view",
              "type": "hotel",
            },
            Object {
              "bedrooms": 2,
              "city": Object {
                "location": Object {
                  "latitude": 51.225402,
                  "longitude": 6.776314,
                  "zoom": 13,
                },
                "name": "Amsterdam",
              },
              "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
              "goods": Array [
                "Breakfast",
                "Washer",
                "Laptop friendly workspace",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 3,
              "images": Array [
                "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
                "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
              ],
              "isFavorite": false,
              "isPremium": true,
              "location": Object {
                "latitude": 51.214402,
                "longitude": 6.764314000000001,
                "zoom": 16,
              },
              "maxAdults": 8,
              "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
              "price": 393,
              "rating": 2.7,
              "title": "Waterfront with extraordinary view",
              "type": "hotel",
            },
          ]
        }
        small={true}
      />
    </div>
  </li>
</ul>
`;
